pandas/core/internals.py

```diff
@@ -1821,6 +1824,130 @@ def _unstack(self, unstacker_func, new_columns):
         return blocks, mask
 
 
+class ExtensionBlock(NonConsolidatableMixIn, Block):
+    """Block for holding extension types.
+
+    Notes
+    -----
+    This is the holds all 3rd-party extension types. It's also the immediate
+    parent class for our internal extension types' blocks, CategoricalBlock.
+
+    All extension arrays *must* be 1-D, which simplifies things a bit.
+    """
+    # Some questions / notes as comments, will be removed.
+    #
+    # Currently inherited from NCB. We'll keep it around until SparseBlock
+    # and DatetimeTZBlock are refactored.
+    # - set
+    # - iget
+    # - should_store
+    # - putmask
+    # - _slice
+    # - _try_cast_result
+    # - unstack
+
+    # Think about overriding these methods from Block
+    # - _maybe_downcast: (never downcast)
+
+    # Methods we can (probably) ignore and just use Block's:
+
+    # * replace / replace_single
+    #   Categorical got Object, but was hopefully unnescessary.
+    #   DatetimeTZ, Sparse got Block
+    # * is_view
+    #   Categorical overrides to say that it is not.
+    #   DatetimeTZ, Sparse inherits Base anyway
+
+    is_extension = True
+
+    # XXX
+    # is_bool is is a change for CategoricalBlock. Used to inherit
+    # from Object to infer from values. If this matters, we should
+    # override it directly in CategoricalBlock so that we infer from
+    # the categories, not the codes.
+    is_bool = False
+
+    def __init__(self, values, placement, ndim=None, fastpath=False):
+        self._holder = type(values)
+        super(ExtensionBlock, self).__init__(values, placement, ndim=ndim,
+                                             fastpath=fastpath)
+
+    def get_values(self, dtype=None):
+        # ExtensionArrays must be iterable, so this works.
+        values = np.asarray(self.values)
+        if values.ndim == self.ndim - 1:
+            values = values.reshape((1,) + values.shape)
```

Comment:
Is this needed? I know it currently like that in `NonConsolidatableBlock`, but do we ever expect the result to be a 2D array if this is holded in an DataFrame. 
Eg datetimetz block returns here a DatetimeIndex for both `.values` and `.get_values()`. On the other hand, a categorical block does this reshaping and returns Categorical vs 2d object numpy array.

Further, do we need to do something with `dtype` arg?

Reply:
this is for sparse. which is a step-child ATM. has to be dealt with